bindings/python/tests/fingerprint_test.py

```diff
@@ -4,37 +4,34 @@
 
 
 class GenerateFingerprintTest(unittest.TestCase):
-    @unittest.skip("https://github.com/arvidn/libtorrent/issues/5985")
     def test_generate(self) -> None:
         # full version
         self.assertEqual(
-            lt.generate_fingerprint_bytes(b"ABCD", 1, 2, 3, 4),  # type: ignore
+            lt.generate_fingerprint_bytes(b"ABCD", 1, 2, 3, 4),
             b"-AB1234-",
         )
 
         # short name
         self.assertEqual(
-            lt.generate_fingerprint_bytes(b"A", 1, 2, 3, 4),  # type: ignore
-            b"-A\x001234-",
+            lt.generate_fingerprint_bytes(b"A", 1, 2, 3, 4),
+            b"---1234-",
```

Comment:
this doesn't look right. what happened to the `A`?

Reply:
The original test was wrong. The behavior of `generate_fingerprint_bytes()` is analogous to `generate_fingerprint()` as expected.

The tests I wrote with `@unittest.skip()` were not all correct (because they wouldn't run, so I couldn't verify the desired condition would be correct until the fix was made).

I expect that here:
 * I wrote and validated the test for `fingerprint()` (`str(fingerprint("A", 1, 2, 3, 4)) == "-A\x001234-"`)
 * I copied that output into some test cases for `generate_fingerprint()`
 * I realized we should have `generate_fingerprint_bytes()`
 * I split the functions into `test_generate()` (with `@unittest.skip()`) and `test_generate_str()`
 * I validated `test_generate_str()` (`generate_fingerprint("A", 1, 2, 3, 4) == "---1234-"`, unlike `fingerprint()`)
 * I forgot to analogously update `test_generate()`

Reply:
I expect the "A" that's passed it would be used in the final fingerprint. It looks like a bug that it turns into `--` instead. I can take a closer look at what's going on

Reply:
I see, there's a check and the `name` *has* to be two bytes long